proposed/middleware-meta.md

```diff
@@ -0,0 +1,133 @@
+PSR-N Meta Document
+===================
+
+1. Summary
+----------
+
+The purpose of this PSR is to provide an interface that defines the formal
+method signature of a PSR-7 compliant HTTP middleware component based on
+existing informal standards.
+
+2. Why Bother?
+--------------
+
+The general concept of a reusable request/response middleware was present in
+[StackPHP][stackphp]. Since the release of PSR-7, a number of frameworks have
+applied the StackPHP concept with a different interface signature. The signature
+for these middleware implementations has been almost universally the same.
+
+Agreeing on a formal middleware interface eliminates several problems and
+provides a number of benefits:
+
+* Provides a formal standard for middleware developers to commit to.
```

Comment:
These goals make sense to you and I know what you're getting at, but the trouble is they're likely not clear to the masses. 

It _almost_ looks like you're defining a definition with the definition. 

"Since PSR-7 has standardized an interface for HTTP requests and responses, we have started to see a huge number of varying implementations of middlewares utilizing these interfaces. Whilst the message interfaces are a great step towards middleware, the actual middleware handlers need to be standardized so that these handlers can be shared between multiple implementations."


- Provide a formal standard for the structure of middleware "handlers".
- Eliminates duplication of multiple incredibly similar interfaces defined by various frameworks.
- Allow sharing of middlewares without minor discrepancies in method signatures breaking things.
- Enables any middleware component to run in any compatible framework.

Or something. 

It's super important to keep language accessible and clear, so we don't have people - who would otherwise be onboard - arguing against it. This happens a lot :D